src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -784,7 +785,7 @@ void SafepointSynchronize::print_safepoint_timeout() {
 
   // To debug the long safepoint, specify both AbortVMOnSafepointTimeout &
   // ShowMessageBoxOnError.
-  if (AbortVMOnSafepointTimeout) {
+  if (AbortVMOnSafepointTimeout && Management::ticks_to_ms(os::elapsed_counter()) > (jlong)AbortVMOnSafepointTimeoutDelay) {
```

Comment:
Rather than use the Management API here perhaps just:
```
if (AbortVMOnSafepointTimeout
    && (os::elapsedTime() * MILLIUNITS > AbortVMOnSafepointTimeoutDelay)) {
```
?

Reply:
To my understanding `os::elapsedTime() * MILLIUNITS` will reduce the granularity of the grace period down to a second but I fully agree this is nicer and we don't need millis for this functionality. 
Thank you for the suggestion, @dholmes-ora

Reply:
`os::elapsedTime()` returns a double so has fractional seconds.